# Day 2 In-class Assignment: Order of Magnitude Modeling

### <p style="text-align: right;"> &#9989; **Jessica Parks** </p>
#### <p style="text-align: right;"> &#9989; Caroline, Luke, Thomas, Jack, Richey</p>

## How many mice would it take to fill up this classroom?

### (a.k.a., making order-of-magnitude estimates for fun and profit!)
![Mouse](https://www.jax.org/-/media/jaxweb/images/jax-mice-and-services/mice/web_0009_find-and-order-jax-mice.jpg?h=316&w=467&la=en&hash=D42556D3A616EC2E099378DB9CA5403D0C38C1E4)

## Learning Goals 

Why are we asking you to do this?

One type of modeling is called "order-of-magnitude estimation" or "order-of-magnitude modeling."  A physicist would call this a "Fermi Problem" after Enrico Fermi, a physicist who participated in the Manhattan Project and who was famously good at making estimates of this sort.  More specifically, this sort of modeling makes us think about:

* The quantities that should go into our models, and the reasonable values of those quantities.
* How the different parts of our model might interact with one another.
* What a "good enough" answer would be to a particular problem

## The Problem

In this project, your group is going to calculate an answer one of the following questions:

1. How many grams of caffeine get consumed on MSU's campus each week?  
2. How much does the air above Lake Superior weigh?
3. If somebody left one of the bathroom faucets running at the Breslin Center, how long would it take to fill it up with water?  
4. If all of the toilet paper used per week in the United States were rolled out across the surface of Michigan could you cover the entire state? If so, could you cover it more than once?
5. Which has more mass, the air in this classroom or the brains in this classroom?  
6. On average, how many people are airborne in the world at a given moment?  
7. How many total miles do all Americans drive in their cars in one year, and how does this compare to the circumference of the Earth ($2.5 \times 10^4$ miles), the distance from the Earth to the Sun ($9 \times 10^7$ miles), and size of the Solar System ($10^9$ miles)?
8. After a particularly exciting Spartan football game, the fans rush the field. How many people can fit onto the football field at Spartan Stadium?

---
# Part 1

**As a group**, you will be assigned one of these problems to answer. You will need to come up with:
1. The questions that you need to ask and answer to come up with an estimate for your problem.
2. The equation(s) you will need to solve your problem.
3. The reasonable values that would go into the equation(s).

Once you have finished these steps, you and your group will implement a simple computer program to calculate a range of answers to your problem.

***Use your whiteboard to write out your ideas so your teammates and instructors can see your ideas/progress***

If you have a hybrid team, try a virtual whiteboard app like [Google Jamboard](https://jamboard.google.com/)


## 1.1 Describing Your Model
**In the cell below, write down your assigned question and the parameters in your model.** Also write down the range of possible values for each of the parameters!



* How many total miles do all Americans drive in their cars in one year, and how does this compare to the circumference of the Earth ($2.5 \times 10^4$ miles), the distance from the Earth to the Sun ($9 \times 10^7$ miles), and size of the Solar System ($10^9$ miles)?*

**Parameters**


* How many miles does the average American drive in one year? *Break down based on oil change*
* How many oil changes does the average American get in a year? *2-3 oil changes*
* How many miles are driven before the next oil change? *3,000 - 5,000 miles*
* What is the average population in the U.S.? *300,000,000 - 400,000,000 people*
* What amount of the population drives? *180,000,000 - 320,000,000 people*


We are saying the average American drives 6,000-15,000 miles.

## 1.2 Coding Your Model
In the cell below, write a program to calculate and print out an estimate for your model.

> **Important**: When writing your code, **define variables for each value of interest.**  Then use those variables in your equations.  In other words, do not  hard-code the values in your equations (do not put *numbers* in your equations, put *variables* in your equations).  
>
> As an example, if I want to compute and then print the area of a right triangle, which I know to be $A = \frac{1}{2} b h$, I might write my code like so:
>
>> ```
>> base = 10
>> height = 5
>>
>> area = (base * height) / 2
>>
>> print("The area is:", area)
>> ```  
>       
> By doing this, if I decide I need to change the `base` value or the `height` value, I can just change the value where they are defined. This can be especially useful if I have to use the values multiple times or if I want to experiment with changing the values.

In [21]:
# write a program in this cell to calculate and print out an estimate
# of the answer to the question you've chosen to solve!

# define variables

population_low = 300000000
population_high = 400000000
miles_low = 3000
miles_high = 5000
oil_change_low = 2
oil_change_high = 3

#identify low and high range

pop_drive_low = population_low * .6
pop_drive_high = population_high * .8

miles_driven_low = miles_low * pop_drive_low * oil_change_low
miles_driven_high = miles_high * pop_drive_high * oil_change_high

print("Low range of miles driven: ", miles_driven_low)
print("High range of miles driven: ", miles_driven_high)


# Compare miles driven to Earth Circumference

earth_circ = 2.5 * (10**4)

circ_low = (pop_drive_low/earth_circ)
circ_high = (pop_drive_high/earth_circ)

print(circ_low, " number of times Americans can drive around the Earth's circumference in a year.")
print(circ_high, " number of times Americans can drive around the Earth's circumference in a year.")

# Compare miles to the sun

dist_to_sun = 9.0 * (10**7)

sun_low = (pop_drive_low/dist_to_sun)
sun_high = (pop_drive_high/dist_to_sun)

print(sun_low, " number of times Americans can drive to the sun in a year.")
print(sun_high, " number of times Americans can drive to the sun in a year")

# Compare size of solar system

solar_system = 1.0 * (10**9)

solar_system_low = (pop_drive_low/solar_system)
solar_system_high = (pop_drive_high/solar_system)

print(sun_low, " number of times Americans can drive across the solar system in a year.")
print(sun_high, " number of times Americans can drive across the solar system in a year")

Low range of miles driven:  1080000000000.0
High range of miles driven:  4800000000000.0
7200.0  number of times Americans can drive around the Earth's circumference in a year.
12800.0  number of times Americans can drive around the Earth's circumference in a year.
2.0  number of times Americans can drive to the sun in a year.
3.5555555555555554  number of times Americans can drive to the sun in a year
2.0  number of times Americans can drive across the solar system in a year.
3.5555555555555554  number of times Americans can drive across the solar system in a year


## 1.3 Exploring Model Results
**Now, run your model several times**, and each time change the values of the parameters to a different value in the range you think is valid.  In the cell below, write down all of your answers.  What is the total range, and approximately what is the average value?

**Range of Miles**

* Low range of miles driven:  1080000000000.0
* High range of miles driven:  4800000000000.0
* Average range of miles driven: 2940000000000

**Comparison to Earth's Circumference**

* 7200.0  number of times Americans can drive around the Earth's circumference in a year.
* 12800.0  number of times Americans can drive around the Earth's circumference in a year.
* Average number of times driven around circumference: 1000

**Comparison to Distance to Sun**


* 2.0  number of times Americans can drive to the sun in a year.
* 3.5555555555555554  number of times Americans can drive to the sun in a year
* 2.775 average number of times American can drive to the sun in a year

**Comparison to Solar System**


* 2.0  number of times Americans can drive across the solar system in a year.
* 3.5555555555555554  number of times Americans can drive across the solar system in a year
* 2.775 average number of times American can drive across the solar system

---
# Part 2

## 2.1 Pick a Second Model
**Now, picking one of the other questions and going through the process of making another order-of-magnitude estimate.** Each group must choose a *different* question. Once you have decided on which question you are going to try to tackle, try working through the answer on your own, but check in with the members of your group if you run into issues or want to bounce ideas off each other.

**Question**

After a particularly exciting Spartan football game, the fans rush the field. How many people can fit onto the football field at Spartan Stadium?

## 2.2 Describe the Model
**In the cell below, write down your chosen question and the parameters in your model.** Also write down the range of possible values for each of the parameters!

* How long is a football field? *120 yards*
* How wide is a football field? *50-55 yards*
* What is the square footage of a person? *8-15 square feet*

## 2.3 Coding the Model

In the cell below, write a program to calculate and print out an estimate for your model.

In [22]:
# write a program in this cell to calculate and print out an estimate
# of the answer to the question you've chosen to solve!

# define variables

sq_footage_low = 8
sq_footage_high = 15

length_field = 120
width_field_low = 50
width_field_high = 55

#define low and high football length

field_low = length_field * width_field_low
field_high = length_field * width_field_high

# identify range of people

people_low = field_low/sq_footage_high
people_high = field_high/sq_footage_low

print("The fewest people on the field possible would be ", people_low)
print("The most people on the field possible would be ", people_high)

The fewest people on the field possible would be  400.0
The most people on the field possible would be  825.0


## Congratulations, you're done!

Submit this assignment by uploading it to the course Desire2Learn web page.  Go to the "In-class assignments" folder, find the appropriate submission link, and upload it there.

See you next class!

&#169; Copyright 2021,  Michigan State University Board of Trustees